lib/cosmos/utilities/store.rb

```diff
@@ -45,14 +44,24 @@ def self.instance(pool_size = 10)
       end
     end
 
-    # Delegate all unknown class methods to delegate to the instance
-    def self.method_missing(message, *args, &block)
-      self.instance.send(message, *args, &block)
-    end
-
-    # Delegate all unknown methods to redis through the @redis_pool
-    ruby2_keywords def method_missing(message, *args, &block)
-      @redis_pool.with { |redis| redis.send(message, *args, &block) }
+    if RUBY_VERSION < "3"
+      # Delegate all unknown class methods to delegate to the instance
+      def self.method_missing(message, *args, &block)
+        self.instance.send(message, *args, &block)
+      end
+      # Delegate all unknown methods to redis through the @redis_pool
+      def method_missing(message, *args, &block)
+        @redis_pool.with { |redis| redis.send(message, *args, &block) }
+      end
+    else
+      # Delegate all unknown class methods to delegate to the instance
+      def self.method_missing(message, *args, **kwargs, &block)
+        self.instance.send(message, *args, **kwargs, &block)
+      end
+      # Delegate all unknown methods to redis through the @redis_pool
+      def method_missing(message, *args, **kwargs, &block)
+        @redis_pool.with { |redis| redis.send(message, *args, **kwargs, &block) }
+      end
```

Comment:
I couldn't get this to work with ruby2_keywords due to the way various redis methods use keyword arguments so I punted and just check for the ruby version.

Reply:
Removes a dependency so 👍